### PR TITLE
Upgrade quickcheck in uint, fixed-hash

### PR DESCRIPTION
**Pull request by kpp**
Fixes https://github.com/paritytech/parity-common/issues/501

### CODE REVIEW

File: uint/src/uint.rs [hunk 1]
```diff
@@ -1761,8 +1758,15 @@ macro_rules! impl_quickcheck_arbitrary_for_uint {
 						$n_bytes / 5
 					};
 
-				let size = g.gen_range(0, range);
-				g.fill_bytes(&mut res[..size]);
+				let size: usize = usize::arbitrary(g) % range;
+
+				let res: [u8; $n_bytes] = $crate::core_::array::from_fn(|i| {
+					if i > range {
```
Comment (ordian) [inline uint/src/uint.rs:1764]:
```suggestion
					if i > size {
```

File: uint/src/uint.rs [hunk 2]
```diff
@@ -1761,8 +1758,15 @@ macro_rules! impl_quickcheck_arbitrary_for_uint {
 						$n_bytes / 5
 					};
 
-				let size = g.gen_range(0, range);
-				g.fill_bytes(&mut res[..size]);
+				let size: usize = usize::arbitrary(g) % range;
```
Comment (ordian) [inline uint/src/uint.rs:1761]:
```suggestion
				let range = $crate::core_::cmp::max(range, 1);
				let size: usize = usize::arbitrary(g) % range;
```

**Review by ordian — APPROVED** [review]
Nice!

File: uint/Cargo.toml [hunk 3]
```diff
@@ -13,16 +13,15 @@ rust-version = "1.56.1"
 [dependencies]
 byteorder = { version = "1.4.2", default-features = false }
 crunchy = { version = "0.2.2", default-features = false }
-qc = { package = "quickcheck", version = "0.9.0", optional = true }
-rand07 = { package = "rand", version = "0.7", default-features = false, optional = true }
+qc = { package = "quickcheck", version = "1", optional = true }
```
Comment (ordian) [inline uint/Cargo.toml:16]: we can also remove the renaming now

File: uint/Cargo.toml [hunk 4]
```diff
@@ -13,16 +13,15 @@ rust-version = "1.56.1"
...
 hex = { version = "0.4", default-features = false }
 static_assertions = "1.0.0"
 arbitrary = { version = "1.0", optional = true }
 
 [features]
 default = ["std"]
 std = ["byteorder/std", "crunchy/std", "hex/std"]
-quickcheck = ["qc", "rand07"]
+quickcheck = ["qc"]
```
Comment (ordian) [inline uint/Cargo.toml:24]:
```suggestion
```

File: uint/Cargo.toml (hunk 3 above)
Comment (ordian) [reply uint/Cargo.toml:16]:
```suggestion
quickcheck = { version = "1", optional = true }
```

File: uint/src/uint.rs (hunk 1 above)
Comment (kpp) [reply uint/src/uint.rs:1764]: Oops

**Pull request merged**